src/tls/boring.rs

```diff
@@ -238,6 +240,37 @@ pub fn grpc_connector(uri: String, root_cert: RootCert) -> Result<TlsGrpcChannel
     Ok(TlsGrpcChannel { uri, client: hyper })
 }
 
+// creating K8s client with boring
+pub async fn create_k8s_client(root_cert: RootCert) -> Result<kube::Client,anyhow::Error> {
+    let mut conn = ssl::SslConnector::builder(ssl::SslMethod::tls_client())?;
+    conn.set_verify(ssl::SslVerifyMode::PEER);
+    conn.set_alpn_protos(Alpn::H2.encode())?;
+    conn.set_min_proto_version(Some(ssl::SslVersion::TLS1_2))?;
+    conn.set_max_proto_version(Some(ssl::SslVersion::TLS1_3))?;
+    match root_cert {
+        RootCert::File(f) => {
+            conn.set_ca_file(f).map_err(Error::InvalidRootCert)?;
+        }
+        RootCert::Static(b) => {
+            conn.cert_store_mut()
+                .add_cert(x509::X509::from_pem(&b).map_err(Error::InvalidRootCert)?)
+                .map_err(Error::InvalidRootCert)?;
+        }
+        RootCert::Default => {} // Already configured to use system root certs
+    }
+    let mut http = hyper::client::HttpConnector::new();
+    http.enforce_http(false);
+    let https = hyper_boring::HttpsConnector::with_connector(http, conn)?;
+    let hyper_client = hyper::Client::builder().http2_only(true).build::<_,hyper::Body>(https);
```

Comment:
Not sure we need http2 only for k8s connection

Reply:
updated